nikola/plugin_categories.py

```diff
@@ -471,3 +472,260 @@ def import_file(self):
     def save_post(self):
         """Save a post to disk."""
         raise NotImplementedError()
+
+
+class Taxonomy(BasePlugin):
+    """Taxonomy for posts.
+
+    A taxonomy plugin allows to classify posts (see #2107) by
+    classification strings. Classification plugins must adjust
+    a set of options to determine certain aspects.
+
+    The following options are class attributes with their default
+    values:
+
+    classification_name = "taxonomy":
+        The classification name to be used for path handlers.
+
+    metadata_name = "taxonomy":
+        The classification name to be used when storing the classification
+        in the metadata. If set to None, the classification won't be stored
+        in the metadata.
+
+    overview_page_variable_name = "taxonomy":
+        Variable for the overview page template which contains the list of
+        classifications.
+
+    more_than_one_classifications_per_post = False:
+        If True, there can be more than one classification per post; in that case,
+        the classification data in the metadata is stored as a list. If False,
+        the classification data in the metadata is stored as a string, or None
+        when no classification is given.
+
+    has_hierarchy = False:
+        Whether the classification has a hierarchy.
+
+    include_posts_from_subhierarchies = False:
+        If True, the list for a classification includes all posts with a
+        sub-classification (in case has_hierarchy is True).
+
+    include_posts_into_hierarchy_root = False:
+        If True, include_posts_from_subhierarchies == True will also insert
+        posts into the list for the empty hierarchy [].
+
+    show_list_as_subcategories_list = False:
+        If not False, for every classification which has at least one
+        subclassification, create a list of subcategories instead of a list/index
+        of posts. This is only used when has_hierarchy = True. If not False, this
+        must be the template name for the list; usually "list.tmpl".
+        If this is set to a string, it is recommended to set
+        include_posts_from_subhierarchies to True to get correct post counts.
+
+    show_list_as_index = False:
+        Whether to show the posts for one classification as an index or
+        as a post list.
+
+    generate_atom_feeds_for_post_lists = False:
+        Whether to generate Atom feeds for post lists in case GENERATE_ATOM is set.
```

Comment:
I've been trying to model the current behavior; there, Atom and RSS are handled pretty differently. Maybe it would make sense to start changing the behavior later? Or should we already add this in this PR?